src/pyvesync/helpers.py

```diff
@@ -169,29 +169,56 @@ def call_api(api: str, method: str,
     @staticmethod
     def check_response(resp: dict, call: str) -> bool:
         common_resp = [
-            'get_devices', '15a_detail', '15a_toggle',
+            'get_devices', '15a_toggle',
             '15a_energy', 'walls_detail', 'walls_toggle',
-            '10a_detail', '10a_toggle', '10a_energy', '15a_ntlight',
+            '10a_toggle', '10a_energy', '15a_ntlight',
             'airpur_detail', 'airpur_status'
         ]
 
+        det_resp = ['15a_detail', '10a_detail']
+
         if isinstance(resp, dict):
-            if call in common_resp:
+            if call in det_resp:
+                if 'code' in resp and resp['code'] == 0:
+                    keys = ['deviceStatus', 'power', 'voltage', 'activeTime']
+                    if all(x in resp for x in keys):
+                        return True
+                    else:
+                        logger.error(
+                            'Keys missing from getting device details')
+                        return False
+                else:
+                    return False
+            elif call in common_resp:
                 if 'code' in resp and resp['code'] == 0:
                     return True
                 else:
+                    logger.error('Error getting details')
                     return False
+
             elif call == 'login' and 'code' in resp:
                 if resp['code'] == 0 and 'result' in resp:
                     return True
                 else:
+                    logger.error('Error getting details')
+                    return False
+            elif call == '7a_detail':
+                keys = ['deviceStatus', 'activeTime',
+                        'energy', 'power', 'voltage']
+                if all(x in resp for x in keys):
+                    return True
+                else:
+                    logger.error('Keys missing from get_details for 7A')
```

Comment:
should there be a return here to match all the other branches?

Reply:
Yes, definitely.  Nice catch